external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -136,6 +137,20 @@ class KafkaTestUtils(
     kdcConf.setProperty(MiniKdc.DEBUG, "true")
     kdc = new MiniKdc(kdcConf, kdcDir)
     kdc.start()
+    val krb5Conf = Source.fromFile(kdc.getKrb5conf, "UTF-8").getLines()
```

Comment:
Maybe we can extract this functionality into a function and this way tests can be added.

Reply:
> Maybe we can extract this functionality into a function and this way simple tests can be added.

Here I want to show add `default_tkt_enctypes` and `default_tgs_enctypes` is useful and it can solve this problem. 

For me, I prefer to build spark's MiniKDC like Kafka and make `krb5.conf` customize. 
Hadoop's MiniKDC  is not so good.

Reply:
> build spark's MiniKDC like Kafka

You mean implement a new SparkKDC?

Reply:
> You mean implement a new SparkKDC?

Yea, I think it's better.

Reply:
> You mean implement a new SparkKDC?

Better to do this.

Reply:
Kafka implemented this because it's not using any Hadoop deps. Here we have a different situation but I think it's not the scope of this PR. On another Jira this can be discussed...

However my previous comment still stands. This can be tested but no tests added. I can personally imagine 2 tests (but maybe one have better idea):
* `s.contains("libdefaults")`
* `!s.contains("libdefaults")`